### DISCUSSION TITLE
Love the project

### DISCUSSION THREAD

**Discussion by IanVicAus** [bartei/rotary-controller-python#21] [General]
Just a quick note to say I love the ideas you are pursuing.
Having a highly configurable DRO would be awesome.
Perhaps a way to also differentiate your product is to make it so that it is easily moved from machine to machine.
For example I have a mill, lathe and grinder and am a hobbyist so are never using all at the one time.
So being able to move the DRO from machine to machine EASILY would be worthwhile - for a charge of say $100 per.
Yes you don't get the $1200 but only $200 (say) but would that be what you get anyway?
For more professional shops the movement has costs so they are less likely to do so but could well want the specialist adaptions that you make available in the DRO.
Anyway all the best and once I get my machines running I look forward to getting a Dr DRO from you.

**Comment by bartei** [comment]
Hi @IanVicAus and thanks for your suggestions!

Implementing a configuration backup restore would probably be enough for what you are describing, it's definitely possible with a few hours of work. I'll add it to the growing list of features!

**Reply by IanVicAus** [reply]
Yes that would be a good way of solving it     YT @ianloy1854

**Comment by IanVicAus** [comment]
Just watched your ELS video, great to the see the project progressing.
A suggestion for the interface - when you select the feed/pitch you display the **selection** and value. But you also have a button to be able to edit them that also shows the selection which is redundant info.  
Suggestion is to make the display button (currently black) also the edit button (maybe a different shade of blue to stay in the realm of selection identification that you use (or clear lines between what would be three selection buttons in a row) and then turn the old edit button into a zeroing button, like you have for all of the axis, and the same shade of blue.
![image](https://github.com/user-attachments/assets/0624de16-d477-4151-9d33-93c78052de76)

Also I would suggest that when in thread mode the two arrow keys are disabled. I can't think of a use case for them to be needed for dynamic usage. They make initial selection really fast - but then risk everything if there is an errant  touch... :( 
It is awesome to have for the feed rate, not quite as interactive as a potentiometer, but FAR better than nothing.
All the best and THANKS for your continuing work.

Last thing is how does it work?
Is it rotary position driving rotary axis.
Or is it rotary position driving desired linear position that is then "followed" by the rotary axis?
The difference between them is one is dependent on the machines accuracy (rotary to rotary)
The other is not dependent on the machine it is dependent on the accuracy of the "electronics"
That is the rotary and linear position and then how accurately the linear position can be followed by the stepper motor.
I am sure it is more work but if it is reasonable I would suggest using the second method.
The reason is that it then negates the issue of worn lead screws, backlash etc as it is using the "actual" position of the cutting tool derived by the linear scale rather than an assumed one by the rotation of the lead screw. Worn everything being an issue for hobbyists.

For absolute accuracy you would want to linear scales on the front and back so that any crabbing of the saddle could be accounted for, and in fact there should be two lead screws.....  I have heard that some CNC machines do it this way now and so don't actually rely on the accuracy of the machine - rather they use the accuracy of the measured position and then adjust the commands to gain the desired output. Though perhaps this is sounding rougher than it is, perhaps they do it with an "accurate" machine and then do this to get a SUPER precision outcome. Basically increase machine precision not through build structure but through position measurement and stepper control.

Trust that all makes sense - I tend to make english hard and that is even without it being read by someone where it is their 2nd (3rd/4th...) language.
